### ISSUE TITLE
Remove hardcoded default values from rules

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#2597]
## Expected Behavior
All the lines of code that we have in our production code are "meaningful".

## Observed Behavior
The second parameter in this functions is never used in production: `valueOrDefault(KEY, "value")` and it's really missleading when you are writing/editing a rule.

## Context
Working on #2498 I found some rules that doesn't have the default value hardcoded in it. And I created commits to fix that (407309756f4886e147eeee233b14d393ed655e8d and e7c0190a0889d05bd4bafa93548a0d058a980391)

But as I continue working in #2498 I see that there are even more rules that have this "issue".

I think that those values are there to make the tests easier. But we should not add any code in production to make the tests easier to write.

My proposal: change the use of `valueOrDefault(key, value)` to `valueString(key)`, `valueBoolean(key)`...

Good thing: This is not a breaking change.

**Comment by arturbosch** [comment]
Those values are the same as in `default-detekt-config`.
If we remove them we need to change the hard dependency on the default-detekt-config or else a programmatically created Rule would'nt be the same as loaded with `--build-upon-default-config`.
Do I miss something?

**Comment by BraisGabin** [comment]
So, for what you are saying, this is wrong, right:

https://github.com/arturbosch/detekt/blob/c5aa152cedb5bf105c2757dda8f4f5e5cf34c405/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/SwallowedException.kt#L66-L77

We should have something like this:

```kotlin
private val ignoredExceptionTypes = SplitPattern(valueOrDefault(IGNORED_EXCEPTION_TYPES,  
  "InterruptedException,NumberFormatException,ParseException,MalformedURLException"))
```

I was confused here because I saw a lot of rules that have this problem. So I didn't know if the default value could be removed or we should fix the rules that doesn't have the proper default values.

**Comment by arturbosch** [comment]
Ah, yes good find. Indeed these properties need the defaults from the configuration documentation.
I would classify this as a bug as users without a config file would not get any issues from these rules.

**Comment by juliabirkett** [comment]
@arturbosch can you share this config documentation so I can take a look at the expected default values?

**Comment by BraisGabin** [comment]
I think that artur refers about this documentation: https://detekt.github.io/detekt/comments.html

**Comment by juliabirkett** [comment]
guys, I just opened my very first PR contributing to Open Source. if you please can take a look and see if it solves this issue 👍
